### PR TITLE
image extract: handle files at the root of a filesystem

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Previously, files like `/app` were incorrectly rejected by faulty
validation logic.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

### CODE REVIEW

**Comment by smarterclayton** [comment]
/hold

Will review but there are a bunch of ugly edge cases in some images that may be impacted with path.clean.

File: pkg/cli/image/extract/extract.go [hunk 1]
```diff
@@ -201,7 +201,7 @@ func parseMappings(images, paths, files []string, requireEmpty bool) ([]Mapping,
 			}
 			mappings = append(mappings, Mapping{
 				Image: image,
-				From:  strings.TrimPrefix(arg, "/"),
+				From:  path.Clean(arg),
```
Comment (soltysh) [inline pkg/cli/image/extract/extract.go:204]: I'd at least start with filepath, which works fine on any OS. path is unix only.

Comment (stevekuznetsov) [reply pkg/cli/image/extract/extract.go:204]: Isn't this the path _in the container_ - so it is going to be a UNIX filesystem? Nobody is doing e.g. `C://` to refer to something in the container FS, are they?

Comment (soltysh) [reply pkg/cli/image/extract/extract.go:204]: What when the container is windows? Obviously it's not here, but I'd prefer to fix it now, rather then later. It's not that hard :wink:

Comment (stevekuznetsov) [reply pkg/cli/image/extract/extract.go:204]: Ah, gotcha

**Comment by stevekuznetsov** [comment]
@smarterclayton can you review this please?

**Pull request closed without merging**